### DISCUSSION TITLE
Cancelation token is a security risk

### DISCUSSION THREAD

**Discussion by Mistofs** [dotnet/runtime#71503] [General] [answered]
You put Cancelation token all over your branches, but you didn't write the implementation of Cancelation token yourself, it's possible that people from dotnet can brick your app on the request of the FBI

**Comment by gfoidl** [comment]
> didn't write the implementation of Cancelation token yourself

But as the project is open source everyone can have a look at the [source code](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/Threading/CancellationToken.cs) and verify what, and how it does to get confidence if needed.

For the shipped binaries one can use a "reflector" like IL-spy to have a look at the built-code to verify that either.
So your stated concern isn't a real one.

Do you have concrete concerns on how someone could use CancellationTokens for mis-use?

**Comment by richlander** [comment, marked as answer]
This post has no technical basis and will only serve to undermine a focus on actual security concerns.

Please don't post issues like this. We have locked this issue and banned this account.

Thanks @gfoidl but there is no reason to engage. This is just a FUD post and clearly not a good faith intent to engage.
